Create HintDb does not erase pre-existing hint db (#19808)
coq/coq: doc/sphinx/proofs/automatic-tactics/auto.rst

.. cmd:: Create HintDb @ident {? discriminated }

>    If a hint database named :n:`@name` already exists, do nothing.
> 
>    Otherwise creates a new hint database named :n:`@ident`. The database is

Reviewer: ```suggestion
   If there is no hint database named :n:`@name`, creates a new hint database
   with that name.  Otherwise, does nothing.  The database is
```

Better to mention the common case before the exceptional case.

   By default, hint databases are undiscriminated.

   .. warn:: @ident already exists and is {? not } discriminated
      :name: mismatched-hint-db

>       `Create HintDb` will not change whether a pre-existing database

Reviewer: ```suggestion
      `Create HintDb` won't change whether an existing database
```
